lightning/src/blinded_path/message.rs

```diff
@@ -84,10 +85,54 @@ impl Writeable for ReceiveTlvs {
 	}
 }
 
+impl Readable for ReceiveTlvs {
+	fn read<R: io::Read>(r: &mut R) -> Result<Self, DecodeError> {
+		_init_and_read_tlv_stream!(r, {
+			(6, path_id, option),
+		});
+		Ok(ReceiveTlvs { path_id })
+	}
+}
+
+/// Represents additional data included by the recipient in a [`BlindedPath`].
+///
+/// This data is encrypted and will be included when sending through
+/// [`BlindedPath`]. The recipient can authenticate the message and
+/// utilize it for further processing if needed.
+#[derive(Clone, Debug)]
+pub enum RecipientData {
+	/// Represents the data specific to [`OffersMessage`]
+	///
+	/// [`OffersMessage`]: crate::onion_message::offers::OffersMessage
+	OffersContext(OffersData),
+	/// Represents the data appended with Custom Onion Message.
+	CustomContext(Vec<u8>),
+}
+
+/// Contains the data specific to [`OffersMessage`]
+///
+/// [`OffersMessage`]: crate::onion_message::offers::OffersMessage
+#[derive(Clone, Debug)]
+pub enum OffersData {
+	/// Payment ID of the outbound BOLT12 payment.
+	WithPaymentId(PaymentId),
+	Empty(()),
```

Comment:
Looks like CI is failing because this needs docs.

Reply:
Introduced documentation and other fuzz test fixes in [e4d4218](https://github.com/lightningdevkit/rust-lightning/commit/e4d4218954dbe9564ed1f2ded56f7af515e2950f)
Thanks for the pointer!